stdlib/char.ml

```diff
@@ -74,3 +76,41 @@ type t = char
 
 let compare c1 c2 = code c1 - code c2
 let equal (c1: t) (c2: t) = compare c1 c2 = 0
+
+(* For bitvects: counting the number of bits of a character *)
+
+(*
+  let bitcount =
+    let bytes = Bytes.create 256 in
+    for c = 0 to 255 do
+      let nbits = ref 0 in
+      for b = 0 to 7 do
+        if (1 lsl b) land c <> 0 then incr nbits
+      done;
+      bytes.[c] <- Char.chr !nbits
+    done;
+    Bytes.unsafe_to_string bytes
+*)
+
+let bitcount =
+  "\000\001\001\002\001\002\002\003\001\002\002\003\002\003\003\004\
+   \001\002\002\003\002\003\003\004\002\003\003\004\003\004\004\005\
+   \001\002\002\003\002\003\003\004\002\003\003\004\003\004\004\005\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \001\002\002\003\002\003\003\004\002\003\003\004\003\004\004\005\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \003\004\004\005\004\005\005\006\004\005\005\006\005\006\006\007\
+   \001\002\002\003\002\003\003\004\002\003\003\004\003\004\004\005\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \003\004\004\005\004\005\005\006\004\005\005\006\005\006\006\007\
+   \002\003\003\004\003\004\004\005\003\004\004\005\004\005\005\006\
+   \003\004\004\005\004\005\005\006\004\005\005\006\005\006\006\007\
+   \003\004\004\005\004\005\005\006\004\005\005\006\005\006\006\007\
+   \004\005\005\006\005\006\006\007\005\006\006\007\006\007\007\008"
```

Comment:
Might be nice to add a sentence saying that this is the output of the command above, and a reference to hacker's delight.

Reply:
Granted for the first point.  As for the reference to Hacker's delight, I don't think it adds anything, considering we are not using one of clever implementations suggested there.

Reply:
Ok, I though it was from there.

Reply:
It seems the "clever" formulas all require 32-bit arithmetics, so I stayed with the naive table-based version.